Zend/tests/anon/gh15994.phpt

```diff
@@ -0,0 +1,11 @@
+--TEST--
+Abstract function must be implemented
+--FILE--
+<?php
+
+$c = new class {
+    abstract public function f();
+};
```

Comment:
Hmm, this message doesn't make too much sense to me, as we cannot implement a method that is also declared as abstract. Maybe we should disallow `abstract` for anonymous classes in the compiler. This message would still make sense when the method comes from a parent class, interface or trait.

Granted, that is not really related to this change, so I'm happy if you move the abstract method to a parent class.

In reality, we should probably also change the error message for this:

```php
class Foo {
    public abstract function bar();
}
```

> Fatal error: Class Foo contains 1 abstract method and must therefore be declared abstract or implement the remaining methods (Foo::bar)

Implementing `Foo::bar()` is a weird suggestion, given the abstract declaration comes from this class.

Reply:
> Maybe we should disallow abstract functions in anonymous classes in the compiler.

I can work on a patch for that

> I'm happy if you move the abstract method to a parent class.

done